src/plugins/data_explorer/public/components/sidebar/index.tsx

```diff
@@ -3,115 +3,95 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useMemo, FC, useEffect, useState } from 'react';
-import { i18n } from '@osd/i18n';
-import {
-  EuiComboBox,
-  EuiSelect,
-  EuiComboBoxOptionOption,
-  EuiSpacer,
-  EuiSplitPanel,
-  EuiPageSideBar,
-} from '@elastic/eui';
+import React, { FC, useEffect, useState, useCallback } from 'react';
+import { EuiSplitPanel, EuiPageSideBar } from '@elastic/eui';
 import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
-import { useView } from '../../utils/use';
 import { DataExplorerServices } from '../../types';
 import { useTypedDispatch, useTypedSelector, setIndexPattern } from '../../utils/state_management';
-import { setView } from '../../utils/state_management/metadata_slice';
+import { DataSourceGroup, DataSourceSelectable, DataSourceType } from '../../../../data/public';
+import { DataSourceOption } from '../../../../data/public/';
 
 export const Sidebar: FC = ({ children }) => {
   const { indexPattern: indexPatternId } = useTypedSelector((state) => state.metadata);
   const dispatch = useTypedDispatch();
-  const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
-  const [selectedOption, setSelectedOption] = useState<EuiComboBoxOptionOption<string>>();
-  const { view, viewRegistry } = useView();
-  const views = viewRegistry.all();
-  const viewOptions = useMemo(
-    () =>
-      views.map(({ id, title }) => ({
-        value: id,
-        text: title,
-      })),
-    [views]
-  );
+  const [selectedSources, setSelectedSources] = useState<DataSourceOption[]>([]);
+  const [dataSourceOptionList, setDataSourceOptionList] = useState<DataSourceGroup[]>([]);
+  const [activeDataSources, setActiveDataSources] = useState<DataSourceType[]>([]);
 
   const {
     services: {
-      data: { indexPatterns },
+      data: { indexPatterns, dataSources },
       notifications: { toasts },
+      application,
     },
   } = useOpenSearchDashboards<DataExplorerServices>();
 
   useEffect(() => {
     let isMounted = true;
-    const fetchIndexPatterns = async () => {
-      await indexPatterns.ensureDefaultIndexPattern();
-      const cache = await indexPatterns.getCache();
-      const currentOptions = (cache || []).map((indexPattern) => ({
-        label: indexPattern.attributes.title,
-        value: indexPattern.id,
-      }));
-      if (isMounted) {
-        setOptions(currentOptions);
+    const subscription = dataSources.dataSourceService.dataSources$.subscribe(
+      (currentDataSources) => {
+        if (isMounted) {
+          setActiveDataSources(Object.values(currentDataSources));
+        }
       }
-    };
-    fetchIndexPatterns();
+    );
 
     return () => {
+      subscription.unsubscribe();
       isMounted = false;
     };
-  }, [indexPatterns]);
+  }, [indexPatterns, dataSources]);
+
+  const getMatchedOption = (dataSourceList: DataSourceGroup[], ipId: string) => {
```

Comment:
There is potential for mistakes here. Let's add some positive and negative tests.